spanner/google/cloud/spanner_v1/snapshot.py

```diff
@@ -194,6 +209,141 @@ def execute_sql(self, sql, params=None, param_types=None, query_mode=None):
         else:
             return StreamedResultSet(iterator)
 
+    def partition_read(self, table, columns, keyset, index='',
```

Comment:
why do we need to make these public here as well? If users need to call these methods they should be using BatchTransaction.

Reply:
They are lower-level, but are API wrappers just like all the other snapshot methods.

Reply:
But we do not want people to be invoking these method directly via a snapshot. They should create a BatchTransaction if they want to invoke these methods. Also these methods are a very niche use case. Why enhance the Snapshot API with methods that hardly anyone will use?

Reply:
As with the `Session` class, or explicit transaction management, we don't expect "normal" users to interact directly with these methods:  we document the recommended patterns instead.

Leaving these methods public, like the other API method wrappers, enables users who have use cases we don't anticpate (e.g. perhaps the kind of fan-out/fan-in I originally sketched) to design new solutions using them as primitives.

Reply:
We can always make these public if needed later based on customer feedback. I think in this specific case, making these methods public might cause more people to directly use these rather than batch transaction since they would be familiar with the Snapshot interface and using it everywhere else.

Reply:
I am generally in favor of making things private until we need them to be public.

Reply:
Done in dbabfba